src/utils.py

```diff
@@ -1267,16 +1268,34 @@ def update_config(args: Dict[str, Any], default_seed: int = 123) -> None:
             # pass in a value and this key is already in the
             # configuration dict, add the current value to args.
             args[k] = getattr(CFG, k)
-    # Maintain the invariant that CFG has some seed and some
-    # experiment_id set. This is very useful in unit tests, where
-    # there are often no command line args being passed.
-    args["seed"] = args.get("seed", default_seed)
-    args["experiment_id"] = args.get("experiment_id", "")
-    for d in [GlobalSettings.get_arg_specific_settings(args), args]:
+    for d in [arg_specific_settings, args]:
         for k, v in d.items():
             CFG.__setattr__(k, v)
 
 
+def reset_config(args: Dict[str, Any], default_seed: int = 123) -> None:
+    """Reset to the default CFG, overriding with anything in args.
+
+    This utility is meant for use in testing only.
+    """
+    parser = create_arg_parser()
+    default_args = parser.parse_args([
+        "--env",
+        "default env placeholder",
+        "--seed",
+        str(default_seed),
+        "--approach",
+        "default approach placeholder",
+    ])
+    arg_dict = {
+        k: v
+        for k, v in GlobalSettings.__dict__.items() if not k.startswith("_")
```

Comment:
Will this work correctly with arg-specific settings?

Reply:
yeah, the arg-specific settings are added in the call to `update_config` at the end of this function